src/logging.c

```diff
@@ -293,6 +293,44 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
     }
 
     g_rcutils_logging_initialized = true;
+
+    // We load the logging configs after setting g_rcutils_logging_initialized
+    // to true otherwise rcutils_logging_set_logger_level will cause recursive
+    // call to this function due to RCUTILS_LOGGING_AUTOINIT Check for the
+    // Check for the environment variable for selecting logging level
+    const char * logging_config_filename;
+    ret_str = rcutils_get_env("RCUTILS_LOGGING_CONFIG_FILE", &logging_config_filename);
+    if (NULL == ret_str && strcmp(logging_config_filename, "") != 0) {
+      FILE * logging_config_file = fopen(logging_config_filename, "r");
+      if (NULL == logging_config_file) {
+        RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+          "Failed to open logging config file `[%s]`.",
+          logging_config_filename);
+        return RCUTILS_RET_ERROR;
+      }
+      char logger_name[50];
+      char severity[10];   // fatal error debug info warn case insensitive
```

Comment:
how about allocate memory dynamically? I think that how it is done when parsing the arguments logging level.

Reply:
Good point, I did some googling, looks like having scanf allocate memory is a [posix extension](https://stackoverflow.com/questions/11926883/scanf-dynamic-allocation) which mean that it may not work on windows (I'm not sure)